pkg/reconciler/v1alpha1/autoscaling/kpa/resources/service.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2019 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"github.com/knative/pkg/kmeta"
+	"github.com/knative/serving/pkg/apis/autoscaling"
+	pav1alpha1 "github.com/knative/serving/pkg/apis/autoscaling/v1alpha1"
+	sv1a1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	kpaLabelKey     = autoscaling.GroupName + "/kpa"
+	metricsPortName = "metrics"
+	// metricsPort is the external port of the service for metrics.
+	metricsPort = int32(9090)
```

Comment:
This feels like it should be using a constant from `pkg/queue`

Reply:
I don't think there is such a constant there.
The possible sources are:
- pkg/apis/serving/v1alpha1/revision_lifecycle.go
- pkg/reconciler/v1alpha1/revision/resources/constants.go

Reply:
There is!

`v1alpha1.RequestQueueMetricsPort` in `revision_lifecycle.go`, no?

Reply:
Decided to go with revision_lifecycle one for now.
At some point I am going to do somehting about this.